### DISCUSSION TITLE
How can I emulate 3+ partitions?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18113] [Q&A] [answered]
As dagster only support multi-partitions definitions with 2 partitions definitions, is there other options to do if I need 3 partitions?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700184541779429?thread_ts=1700184541.779429&cid=C01U954MEER).*

**Comment by clairelin135** [comment, marked as answer]
You could define a `StaticPartitionsDefinition` that represents a 2-dimension partitions def, and use it as a dimension of your `MultiPartitionsDefinition`.

For example, if you had a `StaticPartitionsDefinition` with keys `a` and `b` and a `TimeWindowPartitionsDefinition`, you could compress these two into a `StaticPartitionsDefinition` with the following keys:
```
2023-01-01/a
2023-01-01/b
2023-01-02/a
2023-01-02/b
...
```
Then you can define this "compressed" static partitions def as a dimension of a `MultiPartitionsDefinition`
